api/src/main/java/eu/darkbot/api/managers/QuestAPI.java

```diff
@@ -0,0 +1,54 @@
+package eu.darkbot.api.managers;
+
+import org.jetbrains.annotations.Nullable;
+
+import eu.darkbot.api.API;
+
+/**
+ * Provides access to data for the Quests in-game.
+ */
+public interface QuestAPI extends API.Singleton {
+
+    /**
+     * @return The quest that is currently active and in view.
+     *         Even if there are several accepted quests, only the selected one is
+     *         returned.
+     */
+    @Nullable Quest getCurrenQuest();
+
+    /**
+     * Quest Object
+     */
+    interface Quest {
+
+        /**
+         * @return The Quest ID
+         */
+        int getCurrentQuestId();
```

Comment:
If this is already within a quest, wouldn't it be the specific quest id, not the "current" quest id?
if so, it should just be:
```suggestion
        int getId();
```